File: c/prime_number/prime_number.c

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void usage()
{
    printf("usage: prime_number [number]\n\n");
}

long long check_prime_number(long number)
{
    long i, max = number / 2;

    for (i=2; i <= max; i++) {
        if (number % i == 0) {
            // no prime number
            return 0;
        }

    }

    // prime number
    return 1;
}

int main(int argc, char *argv[])
{
    long long i, min, max;

    if (argc < 2) {
        usage();
        return 0;
    } else if (argc == 2) {
        min = 2;
        max = atoll(argv[1]);
    } else {
        min = atoll(argv[1]);
        max = atoll(argv[2]);
    }

    min = (min < 2) ? 2 : min;

    for (i=min; i <= max; i++) {
        if (check_prime_number(i)) {
            printf("%lld\n", i);
        }
    }

    return 0;
}

File: python/prime_number/prime_number.py
#!/usr/bin/python

import sys

def check_prime_number(number):
    max = number / 2
    i = 2

    while (i <= max):
        if (number % i) == 0:
            return 0
        i += 1

    return 1

def usage():
    print("usage: prime_number [number]")
    sys.exit()

def main():
    min = 2
    max = 100

    if len(sys.argv) < 2:
        usage()
    elif len(sys.argv) == 2:
        min = 2
        max = int(sys.argv[1])
    elif len(sys.argv) == 3:
        min = int(sys.argv[1])
        max = int(sys.argv[2])
    else:
        usage()

    j = min
    while (j <= max):
        if (check_prime_number(j)):
            print(str(j))
        j += 1

main()
